File: app/src/main/java/me/darshpratap/coaster/api/PageInsightService.java
package me.darshpratap.coaster.api;

import android.util.Log;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class PageInsightService {
    private PageInsightService() {}
    private static PageIngsightApi api = null;

    public static PageIngsightApi getInsightApi() {
        if (api == null)  {
            OkHttpClient okHttpClient = new OkHttpClient.Builder()
                    .connectTimeout(1, TimeUnit.MINUTES)
                    .readTimeout(30, TimeUnit.SECONDS)
                    .writeTimeout(15, TimeUnit.SECONDS)
                    .build();

            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl("http://192.168.1.7:3000/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(okHttpClient)
                    .build();

            api = retrofit.create(PageIngsightApi.class);
        }
        Log.d("fuck", "error" + api);
        return api;
    }
}

File: app/src/main/java/me/darshpratap/coaster/api/PageInsightService.kt
package me.darshpratap.coaster.api

import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object PageInsightService {
    private var api: PageInsightApi? = null

    fun getInsightApi(): PageInsightApi? {
        if (api == null) {
            val okHttpClient: OkHttpClient = OkHttpClient.Builder()
                    .connectTimeout(1, TimeUnit.MINUTES)
                    .readTimeout(1, TimeUnit.MINUTES)
                    .writeTimeout(15, TimeUnit.SECONDS)
                    .build()

            val retrofit: Retrofit = Retrofit.Builder()
                    .baseUrl("https://coaster-backend.appspot.com/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(okHttpClient)
                    .build()

            api = retrofit.create(PageInsightApi::class.java)
        }
        return api
    }
}